rsmtool/test_utils.py

```diff
@@ -700,3 +706,251 @@ def check_subgroup_outputs(output_dir, experiment_id, subgroups, file_format='cs
+        try:
+            assert existing_output_path.exists()
+        except AssertionError:
+            sys.stderr.write("\nNo existing output for \"{}\". "
+                             "Creating directory ...\n".format(source))
+            existing_output_path.mkdir(parents=True)
+
+        # get a comparison betwen the two directories
+        dir_comparison = dircmp(updated_output_path, existing_output_path)
+
+        # if no output was found in the updated outputs directory, that's
+        # likely to be a problem so save that source
+        if not dir_comparison.left_list:
+            self.missing_or_empty_sources.append(source)
+            return
+
+        # first delete the files that only exist in the existing output directory
+        # since those are likely old files from old versions that we do not need
+        existing_output_only_files = dir_comparison.right_only
+        for file in existing_output_only_files:
+            remove(existing_output_path / file)
+
+        # Next find all the NEW files in the updated outputs but do not include
+        # the config JSON files or the OLS summary files or
+        # the evaluation/prediction/model files for SKLL models
+        new_files = dir_comparison.left_only
+        excluded_suffixes = ['_rsmtool.json', '_rsmeval.json', '_ols_summary.txt']
+        new_files = [f for f in new_files if not any(f.endswith(suffix) for suffix in excluded_suffixes)]
+        if skll:
+            new_files = [f for f in new_files if not self.is_skll_excluded_file(f)]
+
+        # next we get the files that have changed and try to figure out if they
+        # have actually changed beyond a tolerance level that we care about for
+        # tests. To do this, we run the same function that we use when comparing
+        # the files in the actual test. However, for non-tabular files, we just
+        # assume that they have really changed since we have no easy way to compare.
+        changed_files = dir_comparison.diff_files
+        really_changed_files = []
+        for changed_file in changed_files:
+            include_file = True
+            updated_output_filepath = updated_output_path / changed_file
+            existing_output_filepath = existing_output_path / changed_file
+            file_format = updated_output_filepath.suffix.lstrip('.')
+            if file_format in ['csv', 'tsv', 'xlsx']:
+                try:
+                    check_file_output(str(updated_output_filepath),
+                                      str(existing_output_filepath),
+                                      file_format=file_format)
+                except AssertionError:
+                    pass
+                else:
+                    include_file = False
+
+            if include_file:
+                really_changed_files.append(changed_file)
+
+        # Copy over the new files as well as the really changed files
+        new_or_changed_files = new_files + really_changed_files
```

Comment:
Why don't we  then have `self.updated_files` and `self.new_files`?

Reply:
You mean ALSO have `self.new_files` In addition to `self.updated_files`?

Reply:
Yes: `self.new_files = new files` and `self.updated_files = `really_changed_files`. I don't feel too strongly about it though.

Reply:
That's a good idea. I can do that.

Reply:
Actually, I'd rather not do that since that requires a lot of changes all over the code.

Reply:
That's fine - I'll file an issue for when we feel like it in future